music21/musicxml/m21ToXml.py

```diff
@@ -3364,7 +3366,18 @@ def _spannerEndParameters(spannerClass, sp):
 
         return post
 
-    def objectAttachedSpannersToNotations(self, obj, objectSpannerBundle=None):
+    def getArpeggioNumber(self, arp: expressions.ArpeggioMarkSpanner) -> int:
+        arpeggioNumber: int = self.arpeggioNumbers.get(arp, -1)
+        if arpeggioNumber == -1:
+            arpeggioNumber = self.nextArpeggioNumber
+            self.nextArpeggioNumber += 1
+            if self.nextArpeggioNumber > 16:
+                self.nextArpeggioNumber = 1
+            self.arpeggioNumbers[arp] = arpeggioNumber
+
+        return arpeggioNumber
+
+    def objectAttachedSpannersToNotations(self, obj, noteIndexInChord, objectSpannerBundle=None):
```

Comment:
We have permission to break the signature in a major version change, but I'm wondering if we can't just make this new param keyword-only and add a default value like 0?

Reply:
That seems awkward (it would be the only noteIndexInChord treated this way in the file). How about giving it a default?  The only thing broken then is calls with positional spannerBundle params (which now must be called with a name).

Reply:
I've pushed what this would look like, so you can review it.

Reply:
looks good to me.